app/grandchallenge/evaluation/models.py

```diff
@@ -1143,6 +1158,7 @@ def update_combined_ranks_cache(self):
         combined_ranks = []
         num_phases = self.public_phases.count()
 
+        now = timezone.now()
         for user, evaluations in self.users_best_evaluation_per_phase.items():
             if len(evaluations) == num_phases:  # Exclude missing data
```

Comment:
Maybe not part of this pr perse, but isn't this incorrect? What if a challenge has 3 public phases, and a combined leaderboard for 2 phases (2+3). If a user submitted to all 3 phases, their results are excluded I think?

Reply:
The `public_phases` are property of `CombinedLeaderboard` and hence are scoped to linked phases for the board, not the challenge as a whole.

Reply:
Kicking results out if one of the phases is not public is a stop-gap to prevent result leakage. Maybe a warning should be added for the challenge admins only that signifies this on the leaderboard detail page?

Reply:
Ah, sorry, my bad.

Reply:
> If a user submitted to all 3 phases, their results are excluded I think?

Shouldn't be an issue. `self.users_best_evaluation_per_phase` is also limited to `self.public_phases` (e.g. the phases we are interested in), so results will not be kicked out if the user has submitted to additional public/private phases.